### PR TITLE
Providing a better error message when project references are missing for other files

### PR DESCRIPTION
**Pull request by davidair**
This change improves the error message for issue #4887 for both "pod install" and "pod lib lint": instead of "undefined method `path'", it will actually report a problematic file with "Unable to find other source ref for _path_ for target _target_."

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Better error message for issue 4887  
  [davidair](https://github.com/davidair)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
@davidair thanks! Can you add a changelog entry and a test in `pod_target_installer_spec.rb`?

**Comment by davidair** [comment]
I've updated the CHANGELOG and added the test. Travis CI is failing due to a Danger error, I am assuming it's not related to my changes.

**Comment by dnkoutso** [comment]
Hm, weird is this rebased off of latest master from upstream? @orta thoughts?

**Comment by orta** [comment]
This is because the image is being treated as text in the check for `xit` - which fails, that check ideally should only happen on ruby files

**Comment by orta** [comment]
You can ignore the warning if you want

**Comment by davidair** [comment]
Is the issue happening because spec/fixtures/minions-lib is committed as an expanded folder, as opposed to a single tarball (like spec/fixtures/banana-lib)? If this is the case, I can re-package as a tarball (and add the spec/fixtures/minions-lib to .gitignore)

**Comment by dnkoutso** [comment]
@orta I cannot merge since the warning is breaking one of the builds...

**Comment by davidair** [comment]
I've re-packaged the fixture as a tarball, but I've realized it doesn't make a difference since Danger picks it up anyway. I tried running danger locally to see if I could tweak Dangerfile to fix this but ran into https://github.com/danger/danger/issues/381

**Comment by dnkoutso** [comment]
@davidair is there anything else other than the image you can use to make this work?

**Comment by davidair** [comment]
@dnkoutso: I was not using the image in my test (only the json file) so just removing it should be enough. Giving it a try.

**Comment by dnkoutso** [comment]
👍 can you also after squash to a single commit?

**Comment by dnkoutso** [comment]
@davidair @orta has fixed the issue can you try with the image? I know you got a green build but still nee to squash to a single commit :)

**Comment by CocoaPodsBarista** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: undefined method `ends_with?` for "CHANGELOG.md":String
Did you mean?  end_with?. Updating the Danger gem might fix the issue. Your Danger version: 5.3.0, latest Danger version: 5.3.3

```
 #  from Dangerfile:5
 #  -------------------------------------------
 #    next unless File.file?(file)
 >    next unless file.ends_with? '.rb'
 #  
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by orta** [comment]
Bah, can you change that to `end_with?` in the Dangerfile please - sorry! Been a while since I've done ruby

**Comment by dnkoutso** [comment]
Will fix now.

**Comment by dnkoutso** [comment]
yay! green!

**Comment by davidair** [comment]
Squashed the commits.

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Dimitris Koutsogiorgas](https://github.com/dnkoutso)
   [#6829](https://github.com/CocoaPods/CocoaPods/pull/6829)
 
+* Provide a better error message when references are missing for non-source files
```
Comment (dnkoutso) [inline CHANGELOG.md]: @davidair this is a nit but can you add two empty spaces at the end of this line? Like the other CHANGELOG entries?

Comment (DanToml) [inline CHANGELOG.md]: I'll fix this when I do the release

**Pull request merged**